src/main/java/io/debezium/connector/yugabytedb/YBClientUtils.java

```diff
@@ -1,5 +1,189 @@
+  }
+  
+  /**
+   * Helper function to get the mapped values for the table to tablet IDs. The function returns a
+   * list in which each element is a pair like Pair<tableID, tabletId>
+   * @param ybClient {@link YBClient} instance
+   * @param tableIds set of table UUIDs for which to find the tablet UUIDs
+   * @return a list containing the pairs where tableID is mapped to tabletIDs
+   */
+  public static List<Pair<String, String>> getTabletListMappedToTableIds(YBClient ybClient, 
+                                                                         Set<String> tableIds) {
+    List<Pair<String, String>> tableToTabletIds = new ArrayList<>();
+    try {
+      for (String tableId : tableIds) {
+          YBTable table = ybClient.openTableByUUID(tableId);
+          tableToTabletIds.addAll(ybClient.getTabletUUIDs(table).stream()
+                  .map(tabletId -> new ImmutablePair<String, String>(tableId, tabletId))
+                  .collect(Collectors.toList()));
+      }
+      Collections.sort(tableToTabletIds, (a, b) -> a.getRight().compareTo(b.getRight()));
+    }
+    catch (Exception e) {
+        LOGGER.error("Error while fetching all the tablets", e);
+        throw new DebeziumException(e);
+    }
+
+    return tableToTabletIds;
+  }
+
+  /**
+   * Helper function to set the checkpoint on YugabyteDB server side
+   * @param ybClient {@link YBClient} instance
+   * @param streamId DB stream ID
+   * @param tableId table UUID
+   * @param tabletId tablet UUID
+   * @param term term of the checkpoint
+   * @param index index of the checkpoint
+   * @param initialCheckpoint flag to indicate whether to start retaining intents
+   * @param bootstrap flag to indicate whether to bootstrap the tablet
+   * @throws Exception if things go wrong
+   */
+  public static void setCheckpoint(YBClient ybClient, String streamId, String tableId, 
+                                   String tabletId, long term, long index, 
+                                   boolean initialCheckpoint, boolean bootstrap) throws Exception {
+    String logFormatString = "Connector setting checkpoint for tablet {} with streamId {} - " 
+                             + "term: {} index: {} initialCheckpoint: {} bootstrap: {}";
+    LOGGER.debug(logFormatString, tabletId, streamId, term, index, initialCheckpoint, bootstrap);
+    ybClient.bootstrapTablet(ybClient.openTableByUUID(tableId), streamId, tabletId, term, 
+                             index, initialCheckpoint, bootstrap);
+  }
+
+  /**
+   * Helper function to get the Debezium style TableId of a table from table UUID
+   * @param ybClient the {@link YBClient} instance
+   * @param table the {@link YBTable} instance
+   * @return the {@link TableId}
+   * @throws Exception if a {@link YBTable} cannot be opened by the client
+   */
+  public static TableId getTableIdFromYbTable(YBClient ybClient, YBTable table) throws Exception {
+    ListTablesResponse resp = ybClient.getTablesList(table.getName());
```

Comment:
To use the `TableInfo` object, we need to make a RPC call to the server to give us all the tables (with the specified table name ofcourse). We are using this way because we need the `pgSchemaName` and `TableInfo` is the only class which has access to `pgSchemaName`.

Now I think this could cause too many calls to the server from the connector.

The other approach I can think of is to add a pgSchemaName property to the `YBTable` object - if we follow this way, then since we already have the cached `YBTable` objects for each tableId we will ultimately reduce our dependency on the server.

However, we can still cache this `ListTablesResponse` once with the information of all the tables in the cluster and then can access it as and when needed. But with this, I don't think storing unnecessary information (which we are not going to use often) on the connector seems to be a good idea.

@vrajat @suranjan What are your thoughts on this?